### PR TITLE
improve "blocky fuzz" rendering, remove branches

### LINKED ISSUE #1458: Blocky fuzz issues
Moving this topic to a dedicated issue.

ceski:

> As an aside, the last time I looked at blocky fuzz with the new resolution code it had some significant artifacts (see the fuzz test wad I made). Maybe it should be tweaked a bit before the next release.

> Fuzz hasn't looked right at some point after this commit: #1217

> Try a few different resolutions while looking at the test pattern that replaces the pistol. Then take a look at the fist and how it looks when it's returning to an idle state after coming to a stop (it shrinks in a strange manner). Even without the fuzz effect, player sprites look a little odd when returning to an idle state (move around a bit and then let the weapon sprite return to center).

rfomin:

> I think it's the aftermath of #1377 Perhaps we should revert it.

> I've tested blocky fuzz more and the test pattern that replaces the pistol doesn't look right at many resolutions, e.g. at 300%

### PR DESCRIPTION
**Pull request by rfomin**
Fix #1458

### CODE REVIEW

**Comment by rfomin** [comment]
Before:
woof0001
After:
woof0001
Test WAD by @ceski-1: [fuzztest.zip](https://github.com/user-attachments/files/17799057/fuzztest.zip)

File: src/r_draw.c [hunk 1]
```diff
@@ -423,50 +426,45 @@ static void R_DrawFuzzColumn_orig(void)
...
+            fullcolormap[6 * 256 + dest[linesize * fuzzoffset[fuzzpos]]];
         dest += linesize; // killough 11/98
 
         ++fuzzpos;
 
-        if (fuzzpos == FUZZTABLE)
-            fuzzpos = 0;
-
+        // Clamp table lookup index.
+        fuzzpos &= (fuzzpos - FUZZTABLE) >> (8 * sizeof(fuzzpos) - 1); // killough 1/99
```
Comment (ceski-1) [inline src/r_draw.c:435]: I've never understood why Killough wrote this line this way. Obfuscation? Anything wrong with something like this?
```c
fuzzpos = (fuzzpos + 1) % FUZZTABLE;
```

File: src/r_draw.c [hunk 2]
```diff
@@ -484,40 +482,47 @@ static void R_DrawFuzzColumn_block(void)
...
 
-        for (int i = 0; i < ny && count - i > 0; i++)
+        // if (count < 0)
+        // {
+        //    lines += count;
+        //    count = 0;
+        // }
+        const int mask = count >> (8 * sizeof(mask) - 1);
+        lines += count & mask;
+        count &= ~mask;
```
Comment (ceski-1) [inline src/r_draw.c:502]: Similar question here.

**Review by ceski-1 — COMMENTED** [review]
Looks really nice now!

**Comment by ceski-1** [comment]
> Test WAD

Here's a newer version with an additional test case: [fuzz3.zip](https://github.com/user-attachments/files/17808231/fuzz3.zip)

File: src/r_draw.c (hunk 1 above)
Comment (rfomin) [reply src/r_draw.c:435]: I think `%` is more expensive than `>>`, `&`. See "bit twidling" topic, e.g. https://graphics.stanford.edu/~seander/bithacks.html#IntegerAbs

I'm not sure there's much benefit with modern CPU/compilers, though. From my tests, this branch is about ~10% faster at best.

**Review by ceski-1 — APPROVED** [review]

**Pull request merged**